### PR TITLE
Added support for mod defined application titles

### PR DESCRIPTION
**Pull request by Mailaender**
it was hard-coded to `OpenRA` before.

### CODE REVIEW

**Comment by pchote** [comment]
NRE when switching to the content installer.

**Comment by Mailaender** [comment]
Can't reproduce.

**Comment by pchote** [comment]
1) Launch Red Alert
2) Press "Manage Content"

File: OpenRA.Game/Game.cs
Comment (pchote) [inline OpenRA.Game/Game.cs]: How about falling back to `$"{metadata.Title - OpenRA}"`, so that we then don't need to change anything in the default mod.yamls?

Comment (pchote) [reply OpenRA.Game/Game.cs]: In any case, we should use "OpenRA - {MOD}" instead of "{MOD} - OpenRA" to be consistent with the OS desktop integration, windows start menu, etc.

**Comment by Mailaender** [comment]
Doesn't crash for me.

File: OpenRA.Game/Game.cs
Comment (Mailaender) [reply OpenRA.Game/Game.cs]: Look at your browser: it doesn't say "Firefox - OpenRA.net" but "OpenRA.net - Firefox" so it is still informational when you have many tabs open in your task bar. I always hate it when applications deviate from that standard, so I won't change it. I also don't see the need to hard-code even more pre-configuration. This also acts as a documentation for the feature.

Comment (pchote) [reply OpenRA.Game/Game.cs]: My point was that this PR is deviating from our standard...

Windows:
Screenshot 2021-07-17 at 14 48 48

macOS:
Screenshot 2021-07-17 at 14 46 45

Linux:
Screenshot 2021-07-17 at 14 52 31

Comment (Mailaender) [reply OpenRA.Game/Game.cs]: And you already see that standard is not a good idea, because the mod title is truncated.

Comment (dragunoff) [reply OpenRA.Game/Game.cs]: I agree with @Mailaender here - "{MOD} - OpenRA" is better than "OpenRA - {MOD}" because the important information is at the front and will be visible when the UI can't fit the full title. I can see a benefit of putting OpenRA at the front for keeping the items together when sorting alphabetically but that's about it :thinking: And IMO the packaging should also change to come in line with this new format.

Comment (pchote) [reply OpenRA.Game/Game.cs]: I would agree if the behaviour when truncated were the only motivating factor.

However, we need to consider the implications of how we present our "brand". I believe that our current naming convention is consistent with the [franchise modding guidelines](https://www.ea.com/games/command-and-conquer/command-and-conquer-remastered/modding-faq) because we present them primarily as being OpenRA, with the game title as a descriptive addon that I consider to fall under "you may state that your Mod is for Command & Conquer and/or for a specific C&C game or expansion pack in the franchise".

I personally feel that we would be violating the spirit of this agreement by deemphasising the OpenRA and presenting them primarily as Red Alert, Tiberian Dawn etc. The fact that they are OpenRA projects *is* the important information, so that really should stay at the front.

Comment (Mailaender) [reply OpenRA.Game/Game.cs]: Changed according to your brand guidelines.

**Review by abcdefg30 — DISMISSED** [review]
Lgtm and works. We could still add the fallback as well but I don't think it's necessary (and this way more people are likely to discover this feature for their mods).

**Comment by pchote** [comment]
> and this way more people are likely to discover this feature for their mods

OpenRA's feature feature/documentation discovery problems are more general than just this case. The amount of duplicated boilerplate that needs to be changed when setting a new mod is already frustrating, and I expect that most people would be using their text editors find/replace feature for most of it (that is what I do).

If we really do consider this to be a problem then the solution is to add a new "Manifest" section to https://docs.openra.net that is automatically maintained with the full list of `IGlobalMetadata` plus the hardcoded fields. Forcing yet another duplicated string that the engine already knows how to generate is a poor compromise.

**Comment by Mailaender** [comment]
I think the existing backfall is sufficient: https://github.com/OpenRA/OpenRA/blob/e201e410f405455c4e507f6c804a83a924214643/OpenRA.Platforms.Default/Sdl2PlatformWindow.cs#L218
Having just one override is enough.

**Comment by reaperrr** [comment]
What's the status here?

I could really use this in a project of mine, but I'm not firm enough in platform/app-related topics to want to make the call of whether to merge this as-is or not.

**Comment by Mailaender** [comment]
Some other third-party mods are waiting for this as well.

**Comment by pchote** [comment]
It looks like my request for automatically setting the window title doesn't have much support, so i'm willing to drop that.

However, before merging, can we please change `ApplicationTitle` to `WindowTitle`? This feature is setting the window title, and has no impact on the application title that is used in e.g. dock or launcher shortcuts.

**Comment by Mailaender** [comment]
Changed it to `WindowTitle`.

File: OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs
Comment (reaperrr) [inline OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs:154]: Before we merge this PR, is this just fixing bitrot or an accidental rebase error?

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs:154]: I put it in a separate commit now. These changes while they look simple often result in me reading through a lot of code in various places before I manage to find the right spot to insert to. This was an accidental find that was highlighted by @OmniSharp.

File: OpenRA.Platforms.Default/Sdl2PlatformWindow.cs
Comment (reaperrr) [inline OpenRA.Platforms.Default/Sdl2PlatformWindow.cs]: For consistency, this method name should be updated to SetWindowTitle as well.

LGTM after that.

Comment (Mailaender) [reply OpenRA.Platforms.Default/Sdl2PlatformWindow.cs]: Renamed it everywhere.

**Review by reaperrr — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/11171a4e9b10a7f8fcdaa961c12bf9b63e02de77...8618c022e0e37d2261758ff314ccea8fa7444b8d)

**Pull request merged**
